core/src/test/java/org/hibernate/ogm/backendtck/queries/QueriesWithEmbeddedCollectionTest.java

```diff
@@ -200,6 +220,66 @@ private static void persist(Object... entities) {
 		session.close();
 	}
 
+	private List<ProjectionResult> asProjectionResults(String projectionQuery) {
+		List<?> results = session.createQuery( projectionQuery ).list();
+		List<ProjectionResult> projectionResults = new ArrayList<ProjectionResult>();
+
+		for ( Object result : results ) {
+			if ( !( result instanceof Object[] ) ) {
+				throw new IllegalArgumentException( "No projection result: " + result );
+			}
+			projectionResults.add( ProjectionResult.forArray( (Object[]) result ) );
+		}
+
+		return projectionResults;
+	}
+
+	private static class ProjectionResult {
```

Comment:
I think I've seen that before somewhere else? Maybe you can extract it into test-utils?